ui_framework/components/modal/confirm_modal.js

```diff
@@ -9,57 +9,98 @@ import { KuiModalBody } from './modal_body';
 import { KuiModalBodyText } from './modal_body_text';
 import { KuiButton } from '../index';
 
-export function KuiConfirmModal({
-    message,
-    title,
-    onCancel,
-    onConfirm,
-    cancelButtonText,
-    confirmButtonText,
-    className,
-    ...rest }) {
-  const ariaLabel = rest['aria-label'];
-  const dataTestSubj = rest['data-test-subj'];
-  return (
-    <KuiModal
-      style={{ 'width': '450px' }}
-      data-tests-subj={ dataTestSubj }
-      aria-label={ ariaLabel }
-      className={ className }
-    >
-      {
-        title ?
-          <KuiModalHeader>
-            <KuiModalHeaderTitle data-test-subj="confirmModalTitleText">
-              { title }
-            </KuiModalHeaderTitle>
-          </KuiModalHeader>
-          : null
-      }
-      <KuiModalBody>
-        <KuiModalBodyText data-test-subj="confirmModalBodyText">
-          { message }
-        </KuiModalBodyText>
-      </KuiModalBody>
+export const CONFIRM_BUTTON = 'confirm';
+export const CANCEL_BUTTON = 'cancel';
 
-      <KuiModalFooter>
-        <KuiButton
-          type="hollow"
-          data-test-subj="confirmModalCancelButton"
-          onClick={ onCancel }
-        >
-          {cancelButtonText}
-        </KuiButton>
-        <KuiButton
-          type="primary"
-          data-test-subj="confirmModalConfirmButton"
-          onClick={ onConfirm }
-        >
-          {confirmButtonText}
-        </KuiButton>
-      </KuiModalFooter>
-    </KuiModal>
-  );
+export const CONFIRM_MODAL_BUTTONS = [
+  CONFIRM_BUTTON,
+  CANCEL_BUTTON,
+];
+
+export class KuiConfirmModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.confirmButton;
+    this.cancelButton;
```

Comment:
I think your intent behind these two lines was to clarify that these are properties of this object. For consistency, I think we should either do the same thing for `this.kuiButton` in the `KuiButton` component class or remove these two lines from here.

Also, if we choose to leave these here, I'd suggest initializing them explicitly to something like `null`. Otherwise their intent is not as clear, IMO.